### ISSUE TITLE
Support for excluding the `links` object

### ISSUE THREAD

**Issue by killpack** [fotinakis/jsonapi-serializers#6]
According to the spec, a resource object *may* have a `links` object, but it need not have one: http://jsonapi.org/format/#document-resource-objects

So it would be nice if there were a way to specify in a serializer whether an object ought to have a `links` object. (From poking through the code, it looks like this object is always present.) Or, since there's already support for overriding the `self_link` and `relationship_related_link` methods, maybe if all of the values in a `links` object are null, it could poof itself out of existence.

Thanks for this gem!

**Comment by fotinakis** [comment]
Actually, I think I already built support for this:
https://github.com/fotinakis/jsonapi-serializers/blob/master/lib/jsonapi-serializers/serializer.rb#L304

Can you try doing this?

```ruby
def links
  nil
end
```

**Comment by killpack** [comment]
Ah! That works for the resource object itself. Buuut, it doesn't appear to work for any `links` objects in the `relationships` object.

Here's a snippet:
```json
...
"relationships": {
  "child-topics": {
    "links": {
      "self":  "/topics/1/relationships/child-topics",
      "related":"/topics/1/child-topics"
    },
    "data":[
      {"type":"topics","id":"2"}
    ]
  }
}
```
(The `topics` serializer contains the fix you provided.)

It looks like I reported the wrong thing. The actual issue is that the `links` object is always present *inside relationships*. For example: https://github.com/fotinakis/jsonapi-serializers/blob/master/lib/jsonapi-serializers/serializer.rb#L117

Per spec, it looks like you only need at least one of `links`, `data`, or `meta`:
http://jsonapi.org/format/#document-resource-object-relationships 

Sorry for the mixup.

**Comment by killpack** [comment]
I guess this isn't actually a big deal, though, and I could foresee this getting pretty complicated pretty quickly.

**Comment by fotinakis** [comment]
Ah gotcha, makes sense. What do you think is the best interface to nullifying relationship links? Add a property like `has_many :child_topics, include_links: false`? Or some other clearer way?

**Comment by killpack** [comment]
That sounds pretty clear to me!

**Comment by fotinakis** [comment]
Ok cool, I've got a busy week coming up but I will try to take a look at this next week. Happy to review a PR if this is urgent to you, thanks for the report.

**Comment by killpack** [comment]
Not urgent at all. Thanks!

**Comment by fotinakis** [comment]
This is now fixed in v0.2.4, if you override relationship_self_link and relationship_related_link to return nil, `links` will be excluded from the serialized object.
